networkx/algorithms/tree/tests/test_operations.py

```diff
@@ -2,10 +2,22 @@
 
 """
 
+from itertools import chain
+
 import networkx as nx
 from networkx.utils import edges_equal, nodes_equal
 
 
+def _check_label_attribute(input_trees, res_tree, label_attribute="_old"):
+    """Tests to check if label_attribute is storing"""
+
+    res_attr_dict = nx.get_node_attributes(res_tree, label_attribute)
+    res_attr_list = list(res_attr_dict.values())
+    input_label = (list(tree[0].nodes()) for tree in input_trees)
+    input_label_list = list(chain.from_iterable(input_label))
+    return res_attr_list == input_label_list
```

Comment:
This looks good -- :}

I wonder if the `check_attrribute` function should use sets instead of lists for the two label lists. Right now the test passes, but I can imagine a case where the order of the labels might differ, but we'd still have the right set of labels. The function would return False, but it should return True.